### PR TITLE
Small Biz

### LINKED ISSUE #20: Update to V4 of the API
According to the SAM API documentation, only v4 of the API is supported. Currently, samwise uses v1.

### PR DESCRIPTION
**Pull request by vdavez**
This would close #20 and #21 and adds a new method `is_small_business?` to the gem.

### CODE REVIEW

**Comment by adelevie** [comment]
Nice! Where/how would we get the NAICS code from the vendor in the micro-purchase platform?

**Comment by vdavez** [comment]
Seems like we have two options: 
1. hard code [54151](http://www.census.gov/cgi-bin/sssd/naics/naicsrch?code=54151&search=2007%20NAICS%20Search), which covers "Computer Systems Design and Related Services", and assume that vendors will have this NAICS code.
2. Include it in the platform as metadata for the auction.

**Comment by vdavez** [comment]
So, I just did a review of all of the winners from the micro-purchase platform, and all of them have _either_ 54151 or [511210](http://www.census.gov/cgi-bin/sssd/naics/naicsrch?code=511210&search=2007). So, it seems like we can hard code to check for both.

**Comment by vdavez** [comment]
Right, so, if you run 

``` ruby
small = client.is_small_business?(duns: duns, naicsCode: 54151) or client.is_small_business?(duns: duns, naicsCode: 511210)
```

you can find out whether the vendor is a small business for each of the current winners (haven't run it against all users in SAM)...

**Comment by vdavez** [comment]
Ok. So I think this is ready for merge. @adelevie, want to do a review?

**Comment by vdavez** [comment]
Successfully ran this against all of the registered users in the micro-purchase platform. Should be good to go.

**Comment by harrisj** [comment]
This looks good to me, but I will leave for @adelevie to review

File: lib/samwise/client.rb [hunk 1]
```diff
@@ -33,6 +33,32 @@ def is_excluded?(duns: nil)
       JSON.parse(response.body)["hasKnownExclusion"] == false
     end
 
+    def is_small_business?(duns: nil, naicsCode: nil)
```
Comment (jessieay) [inline lib/samwise/client.rb]: can we rename to `small_business?` I generally prefer to keep `is` out of method names since it isn't necessary for clarity

**Comment by jessieay** [comment]
one naming comment, otherwise LGTM!

**Comment by jessieay** [comment]
updates look awesome to me -- @adelevie you want to do the honors?

File: lib/samwise/client.rb (hunk 1 above)
Comment (jessieay) [inline lib/samwise/client.rb]: BTW - this naming comment is related to how predicate matchers work in RSpec:

https://www.relishapp.com/rspec/rspec-expectations/v/2-0/docs/matchers/predicate-matchers

``` ruby
expect(object).to be_small_business

expect(object).to be_is_small_business
```

latter sounds more awkward so I tend to name predicate methods in a way that works well with the matchers. That might not be the best reason I've ever come up with for why I do something, but it's a reason!

**Comment by jessieay** [comment]
@vzvenyach you inspired me! https://github.com/bbatsov/ruby-style-guide/pull/566

**Pull request merged**